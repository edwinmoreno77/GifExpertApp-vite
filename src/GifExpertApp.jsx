import { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Goku']);

    const onAddCategory = (newCat) => {
        setCategories([newCat, ...categories]);
    }

    return (
        <div>
            <h1>GifExpertApp</h1>
            <AddCategory onNewCategory={newCat => onAddCategory(newCat)} />
            <ol>
                {categories.map(cat => {
                    return <li key={cat}>{cat}</li>
                })
                }
            </ol>
        </div>
    )
}
