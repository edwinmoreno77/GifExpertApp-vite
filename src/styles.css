*{
    font-family: Helvetica, Arial, sans-serif;
}

body{
    padding: 60px;
    color: aliceblue;
    background-color: rgb(1, 1, 20);
    text-align: center;
    
}

input{
    color: grey;
    font-size: 1.2rem;
    width: 50%;
    border-radius: 0.4rem;
    border-width: 0.2rem;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}
.card{
    align-content: center;
    border: 4px solid white;
    border-radius: 0.3rem;
    margin-bottom: 10px;
    margin-right: 10px;
    overflow: hidden;
    background-color: white;
    
}

.card img {
    max-width: 400px;
    max-height: 205px;
}  

.titulo{
    text-align: center;
}
.p{
    font-size: 0.5rem;
    background-color: white;
    color:black;
}