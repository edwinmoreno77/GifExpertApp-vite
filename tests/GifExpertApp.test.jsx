import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';





describe('GifExpertApp test', () => {

    test('should render correctly', () => {

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        // screen.debug();


    }
    );


});