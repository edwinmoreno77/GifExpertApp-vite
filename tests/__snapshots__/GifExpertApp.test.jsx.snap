// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp test should render correctly 1`] = `
<div>
  <div>
    <h1>
      Gif Expert App
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Search Gif"
        type="text"
        value=""
      />
    </form>
    <ol>
      <h2
        class="animate__animated animate__fadeIn"
      >
        Goku
      </h2>
      <h2
        class="animate__animated animate__flash"
      >
        Loading...
      </h2>
      <div
        class="card-grid"
      />
    </ol>
  </div>
</div>
`;
